main_ISTA.py

```diff
@@ -27,15 +27,25 @@ def __call__(self, algorithm: Algorithm):
         if algorithm.iteration >= self.max_iteration:
             raise StopIteration
 
-
+class MyPreconditioner(Preconditioner):
+    # See:
+    # Tsai, Y.-J., Bousse, A., Ehrhardt, M.J., Stearns, C.W., Ahn, S., Hutton, B., Arridge, S., Thielemans, K., 2017. 
+    # Fast Quasi-Newton Algorithms for Penalized Reconstruction in Emission Tomography and Further Improvements via Preconditioning. 
+    # IEEE Transactions on Medical Imaging 1. https://doi.org/10.1109/tmi.2017.2786865
+    def __init__(self, kappa):
+        self.kappasq = kappa * kappa + 1e-5
```

Comment:
```suggestion
        # add an epsilon to avoid division by zero. This eps value probably should be made dependent on kappa though.
        self.kappasq = kappa * kappa + 1e-5
```